client/my-sites/stats/stats-purchase/index.tsx

```diff
@@ -37,6 +38,7 @@ const StatsPurchasePage = ( { query }: { query: { redirect_uri: string; from: st
 	const siteSlug = useSelector( ( state ) => getSiteSlug( state, siteId ) );
 
 	const siteProducts = useSelector( ( state ) => getSiteProducts( state, siteId ) );
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
```

Comment:
`currencyCode` could originate from several actions:

https://github.com/Automattic/wp-calypso/blob/d70eab72797db701b872e108986a3866fcd5cdde/client/state/currency-code/reducer.js#L12

It seems the value of the state depends on which action comes back last, so I doubt whether this is about whether whether the value is ready or not. Also I could not reproduce the issue itself. IMHO, if the latest `currencyCode` could be populated then it wouldn't be a problem. That said the approach linked above to set `currencyCode` is expected to have problem like this.

Reply:
Indeed, I assumed those actions of `plans`, `products`, and `products-list` should return the same currency code for all items, or there should be a problem with using the first item's currency code as well.

> Also I could not reproduce the issue itself under my account (NZD)

I tested it with sandboxed API for slower response and found the issue when refreshing the page with `commercialProduct` and `pwywProduct` in memory, and the currency code was being determined. It did not occur for every refreshing.

Reply:
Since I'm [changing](https://github.com/Automattic/wp-calypso/pull/79770) to rely only on `commercialProduct` and `pwywProduct`, could we remove `currencyCode` here. I'm also a bit concerned about the logic behind these.